python.cc

```diff
@@ -149,8 +149,8 @@ PyObject* emd_entry(
   C *cache = nullptr;
   std::unique_ptr<C> cache_ptr;
   if (cache_obj != Py_None) {
-    cache = reinterpret_cast<C *>(reinterpret_cast<intptr_t>(
-        PyLong_AsLong(cache_obj)));
+	long  l=PyLong_AsLong(cache_obj);
+	cache = reinterpret_cast<C *>(reinterpret_cast<intptr_t>( &l ));
```

Comment:
This is no good: instead of converting an integer to a cache pointer you pretend that a local variable's address is a cache pointer. The size of `long` on Windows equals to 4 bytes while the pointer size is 8 bytes. Can you please replace `PyLong_AsLong` with `PyLong_AsLongLong` here?

```suggestion
    cache = reinterpret_cast<C *>(reinterpret_cast<intptr_t>(
        PyLong_AsLongLong(cache_obj)));
```

Reply:
I am unable to get this to work. I continue to get `python.cc(154): error C2440: 'reinterpret_cast': cannot convert from '__int64' to 'intptr_t'`

Reply:
I changed `reinterpret_cast<intptr_t>(PyLong_AsLongLong(cache_obj))` to `reinterpret_cast<intptr_t *>(PyLong_AsLongLong(cache_obj))` and it compiled successfully, I'm not exactly sure about the implications of that, though.